driver/bpf/probe.c

```diff
@@ -59,9 +66,6 @@ BPF_PROBE("raw_syscalls/", sys_enter, sys_enter_args)
 	if (!sc_evt)
 		return 0;
 
-	if (sc_evt->flags & UF_UNINTERESTING)
-		return 0;
-
 	if (sc_evt->flags & UF_USED) {
```

Comment:
Unsure about the use case for these settings that are probably legacy leftovers?
https://github.com/falcosecurity/libs/blob/fb22f8eb84c4cccc63d8dd040ccbbdf579899dff/driver/bpf/probe.c#L65-L71

Would it be possible to remove all of the code related to dropping_mode and kernel side random sampling? 
https://github.com/falcosecurity/libs/blob/fb22f8eb84c4cccc63d8dd040ccbbdf579899dff/userspace/libscap/engine/bpf/scap_bpf.c#L1421

For every bpf tail call in sys enter and exit for the interesting syscalls we make an unnecessary call to drop_event https://github.com/falcosecurity/libs/blob/fb22f8eb84c4cccc63d8dd040ccbbdf579899dff/driver/bpf/plumbing_helpers.h#L545

Reply:
FWIW, we do need kernel side sampling and UF_ALWAYS_DROP/UF_NEVER_DROP (or equivalent functionality provided by some other mechanism)

Reply:
Yup, agree with @gnosek .
My aim is for something like https://github.com/falcosecurity/libs/pull/521#issuecomment-1205032197 for the future, ie: removing the dropping logic from the kernel drivers, and instead allow userspace to take full control.
This means:
* much more customizable logic for clients
* much more powerful logic, given that kernel drivers are a much more constrained environment
* much cleaner driver behavior, ie: nothing is hidden to userspace: every event is sent to userspace unless userspace decides it is under pressure and asks sinsp->scap->drivers to set some syscalls/tracepoints as uninteresting

Reply:
yeah agree with @FedeDP  we cannot remove all this stuff in just one step, we will do it in little steps to avoid breaking all :rofl:

Reply:
Totally, sorry should have been more clear about that this is a comment not intended for refactor all in once, but rather to not forget about it 🙃 

@gnosek just out of curiosity what are those use cases for threat detection that require random kernel side sampling?

yes, also think in the future there can be even better mechanisms for more deterministic kernel side dropping / filtering to alleviate unwanted back pressure from userspace. Will retain some suggestions for a future discussion to not overload this discussion here.

Reply:
@FedeDP oops totally overlooked your previous comment, thanks for linking to it.

Reply:
@incertum not for threat detection but for performance monitoring (the threat detection events are never dropped anyway). When we're running under high load, we can drop the less important events to make more room in the ring buffers for the critical ones. The current implementation is basically "drop non-critical events for all but the first 1/nth of every second", with n being a power of two between 1 and 128 (inclusive), set from userspace.

Reply:
Thanks @gnosek , this makes sense for performance monitoring. Hopefully we find the right approach to support both use cases.

For threat detection use cases more system calls should probably have `UF_NEVER_DROP` set:

```
__NR_chmod 
__NR_fchmodat
__NR_fchmod
__NR_mount 
__NR_umount
__NR_umount2
__NR_ptrace
__NR_connect
__NR_accept
__NR_accept4
__NR_symlink
__NR_symlinkat
__NR_setuid
__NR_unshare
__NR_bpf
__NR_seccomp
__NR_openat2
__NR_mprotect
__NR_open_by_handle_at
__NR_setns
__NR_capset
...
```

Reply:
Yeah right now there is quite a mess in our flags, we need to clean them a little bit, but I think this will be a future step :rocket: We need to specify the specific meaning of each one and understand when we have to label a syscall with it! @incertum these are all good points in my opinion maybe we can create an issue on that :thinking: WDYT?